### DISCUSSION TITLE
Rear Port and Front Port

### DISCUSSION THREAD

**Discussion by garciaoliveira** [netbox-community/netbox#17651] [Help Wanted!]
Hello,

Can anyone help me make the following scenario work on netbox?

Interface -> Rear Port -> Front Port -> Front Port -> Rear Port -> Interface

When I try to do this scenario I can't see the interface connections until there is an interface.

Is there any way to solve the problem?

**Comment by candlerb** [comment]
> I can't see the interface connections until there is an interface

What *exactly* are you looking at?

You won't see a result in the "Connection" column in the Interfaces table until it's a complete interface-to-interface connection; that's what "Connection" means. But you will see "Link Peer" as the first hop, and if you click on the cable trace button you should see the partial path so far.

If that's not what your problem is, then show exactly what version of Netbox you're using, the exact steps to recreate the issue, and screenshots to illustrate the issue.

Aside: normally you would connect interface->front port, and rear port->rear port.

**Reply by garciaoliveira** [reply]
Hello,

When I make the connection Interface -> Front -> Rear -> Rear -> Front -> Interface, everything works fine, and the full connection path is displayed.


![diagrama](https://github.com/user-attachments/assets/67ced009-5227-4d33-9f62-7865470f4c58)


But the connection Interface -> Rear Port -> Front Port -> Front Port -> Rear Port -> Interface

![image](https://github.com/user-attachments/assets/0422f354-7f18-4df9-958c-477b67fd9298)

click in the "Front Port BA1-ODFM1-SLB-F01 (Cable #8)"

![image](https://github.com/user-attachments/assets/2648b896-a401-485d-9e22-d95a1b01ea55)

I know that the normal is Interface -> Front -> Rear -> Rear -> Front -> Interface.

But I would like to know if it is possible to obtain the same result with the connection (Interface -> Rear Port -> Front Port -> Front Port -> Rear Port -> Interface)

**Comment by garciaoliveira** [comment]
Hello,

Can someone help me?

**Reply by DanSheps** [reply]
I guess the question is, why cant you go: `Interface <> FP:RP <> RP:FP <> Interface`

**Reply by garciaoliveira** [reply]
The question is, why cant you go: Interface <> RP:FP <> FP:RP <> Interface

**Reply by candlerb** [reply]
Because that's not how frontports and rearports are modelled in Netbox.

Frontport-to-rearport is a many-to-one relationship, i.e. many frontports can be linked to one rearport, in order to model cable bundles which link patch panels - and avoid having to create many individual rearport-to-rearport connections. It was not designed to be used "backwards".

**Reply by garciaoliveira** [reply]
hi,

Thank you for your reply.

But is there no option or workaround to modify this?

**Reply by candlerb** [reply]
The workaround, as suggested by @DanSheps, is to connect cables from interface to frontport, and from rearport to rearport.

To modify this requires changing the data model and the code significantly. You'd need to start by raising a feature request to support this way of working, and then if it were accepted by the Netbox core team, you could work on the code changes and submit them as a PR for review and possible merge.
